### ISSUE TITLE
🐛 C3: `pages:build` script with pnpm scaffolding for Next does not work

### ISSUE THREAD

**Issue by dario-piotrowicz** [cloudflare/workers-sdk#3654]
### Which Cloudflare product(s) does this pertain to?

C3 v.2.0.14

### Describe the Bug

I've created a next application via 
```
pnpm create cloudflare
```

And the resulting package.json does use pnpm for its scripts:
![Screenshot 2023-07-21 at 14 48 13](https://github.com/cloudflare/workers-sdk/assets/61631103/d23873e7-08c7-4c9c-af94-da7016828dfe)

But the `pages:build` script does not work:
![Screenshot 2023-07-21 at 14 51 10](https://github.com/cloudflare/workers-sdk/assets/61631103/18de7cff-97d6-484e-a0fc-59c2306768ba)

The command should be
```
pnpm exec next-on-pages
```

![Screenshot 2023-07-21 at 14 54 27](https://github.com/cloudflare/workers-sdk/assets/61631103/05b9ce33-03ce-4096-996f-ef7810137eea)

(same for `pages:watch`)
___

This is the relevant piece of code that needs updating:

https://github.com/cloudflare/workers-sdk/blob/99baf58b7c35e85c90e1f6df4cea841f31c0a709/packages/create-cloudflare/src/frameworks/next/index.ts#L90-L93

___

I'm opening this issue for pnpm but yarn (with its different versions) also needs to be tested to make sure that the script works properly with it

**Comment by dario-piotrowicz** [comment]
(PS: I've been working on this small utility library: [package-manager-manager](https://www.npmjs.com/package/package-manager-manager), if we'd like to give it a shot I think it should really help here)

**Comment by dario-piotrowicz** [comment]
I just explored the above with all possible pms (except yarn berry)

<details>
<summary>results (showing how things need to change)</summary>

## npm

### Current version
```ts
    "pages:build": "npx @cloudflare/next-on-pages@1",
    "pages:deploy": "npm run pages:build && wrangler pages deploy .vercel/output/static",
    "pages:watch": "npx @cloudflare/next-on-pages@1 --watch",
    "pages:dev": "npx wrangler pages dev .vercel/output/static --compatibility-flag=nodejs_compat"
```

### Correct version
```ts
    "pages:build": "npx @cloudflare/next-on-pages",
    "pages:deploy": "npm run pages:build && wrangler pages deploy .vercel/output/static",
    "pages:watch": "npx @cloudflare/next-on-pages --watch",
    "pages:dev": "npx wrangler pages dev .vercel/output/static --compatibility-flag=nodejs_compat"
```
#### Notes
 - with `@1`, `npx` asks you to install next-on-pages


## pnpm

### Current version
```ts
    "pages:build": "pnpm dlx @cloudflare/next-on-pages@1",
    "pages:deploy": "pnpm run pages:build && wrangler pages deploy .vercel/output/static",
    "pages:watch": "pnpm @cloudflare/next-on-pages@1 --watch",
    "pages:dev": "pnpm wrangler pages dev .vercel/output/static --compatibility-flag=nodejs_compat"
```

### Correct version
```ts
    "pages:build": "pnpm next-on-pages",
    "pages:deploy": "pnpm pages:build && wrangler pages deploy .vercel/output/static",
    "pages:watch": "pnpm next-on-pages --watch",
    "pages:dev": "pnpm wrangler pages dev .vercel/output/static --compatibility-flag=nodejs_compat"
```

#### Notes
 - `pnpm @cloudflare/next-on-pages` does not work, pnpm wants you to remove the `@cloudflare/` scope
 - also `@1` needs to be removed
 - minor: in `pnpm run pages:build`, `run` is unnecessary
 - opposite what I thought, in regards to `--` the scripts here are the same as npm, so there isn't anything to change regarding that

### yarn

### Current version
```ts
    "pages:build": "yarn @cloudflare/next-on-pages@1",
    "pages:deploy": "yarn run pages:build && wrangler pages deploy .vercel/output/static",
    "pages:watch": "yarn @cloudflare/next-on-pages@1 --watch",
    "pages:dev": "yarn wrangler pages dev .vercel/output/static --compatibility-flag=nodejs_compat"
```

### Correct version
```ts
    "pages:build": "yarn next-on-pages",
    "pages:deploy": "yarn pages:build && wrangler pages deploy .vercel/output/static",
    "pages:watch": "yarn @cloudflare/next-on-pages@1 --watch",
    "pages:dev": "yarn wrangler pages dev .vercel/output/static --compatibility-flag=nodejs_compat"
```

#### Notes
 - `yarn` is basically identical to pnpm (but with `yarn` instead of `pnpm`)
 - just tested on yarn classic

### bun

### Current version
```ts
    "pages:build": "bunx @cloudflare/next-on-pages",
    "pages:deploy": "bun run pages:build && wrangler pages deploy .vercel/output/static",
    "pages:watch": "bunx @cloudflare/next-on-pages@1 --watch",
    "pages:dev": "bunx wrangler pages dev .vercel/output/static --compatibility-flag=nodejs_compat"
```

### Correct version
```ts
    "pages:build": "bunx @cloudflare/next-on-pages",
    "pages:deploy": "bun pages:build && wrangler pages deploy .vercel/output/static",
    "pages:watch": "bunx @cloudflare/next-on-pages --watch",
    "pages:dev": "bunx wrangler pages dev .vercel/output/static --compatibility-flag=nodejs_compat"
```

#### Notes
 - basically identical to `npm` (except that `npm` and `npx` become respectively `bun` and `bunx`), with the exception that `run` is optional (but it's not mandatory to omit)
</details>

it looks like what's needed is to have different set of scripts based on 2 conditions:
 - if pm is `npm` or `bun`: npm/bun version (which are the same, except for `run` which we might leave in)
 - if pm is `yarn` or `pnpm`: yarn/pnpm version (which are the same)
 
 (PS: for this use case using package-manager-manager doesn't seem strictly necessary, it's probably better to just fix this now and consider later if pmm can bring something to the table)
